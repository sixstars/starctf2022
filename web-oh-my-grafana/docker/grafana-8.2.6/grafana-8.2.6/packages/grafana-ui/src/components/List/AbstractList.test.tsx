import React from 'react';
import { shallow } from 'enzyme';
import { AbstractList } from './AbstractList';

describe('AbstractList', () => {
  it('renders items using renderItem prop function', () => {
    const items = [
      { name: 'Item 1', id: 'item1' },
      { name: 'Item 2', id: 'item2' },
      { name: 'Item 3', id: 'item3' },
    ];

    const list = shallow(
      <AbstractList
        items={items}
        renderItem={(item) => (
          <div>
            <h1>{item.name}</h1>
            <small>{item.id}</small>
          </div>
        )}
      />
    );

    expect(list).toMatchSnapshot();
  });

  it('allows custom item key', () => {
    const items = [
      { name: 'Item 1', id: 'item1' },
      { name: 'Item 2', id: 'item2' },
      { name: 'Item 3', id: 'item3' },
    ];

    const list = shallow(
      <AbstractList
        items={items}
        getItemKey={(item) => item.id}
        renderItem={(item) => (
          <div>
            <h1>{item.name}</h1>
            <small>{item.id}</small>
          </div>
        )}
      />
    );

    expect(list).toMatchSnapshot();
  });
});
