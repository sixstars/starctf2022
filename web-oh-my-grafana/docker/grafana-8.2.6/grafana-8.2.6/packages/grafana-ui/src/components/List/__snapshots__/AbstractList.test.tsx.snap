// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractList allows custom item key 1`] = `
<ul
  className="css-1ld8h5b"
>
  <li
    className="css-8qpjjf"
    key="item1"
  >
    <div>
      <h1>
        Item 1
      </h1>
      <small>
        item1
      </small>
    </div>
  </li>
  <li
    className="css-8qpjjf"
    key="item2"
  >
    <div>
      <h1>
        Item 2
      </h1>
      <small>
        item2
      </small>
    </div>
  </li>
  <li
    className="css-8qpjjf"
    key="item3"
  >
    <div>
      <h1>
        Item 3
      </h1>
      <small>
        item3
      </small>
    </div>
  </li>
</ul>
`;

exports[`AbstractList renders items using renderItem prop function 1`] = `
<ul
  className="css-1ld8h5b"
>
  <li
    className="css-8qpjjf"
    key="0"
  >
    <div>
      <h1>
        Item 1
      </h1>
      <small>
        item1
      </small>
    </div>
  </li>
  <li
    className="css-8qpjjf"
    key="1"
  >
    <div>
      <h1>
        Item 2
      </h1>
      <small>
        item2
      </small>
    </div>
  </li>
  <li
    className="css-8qpjjf"
    key="2"
  >
    <div>
      <h1>
        Item 3
      </h1>
      <small>
        item3
      </small>
    </div>
  </li>
</ul>
`;
